### PR TITLE
PR pipeline: use 'max_in_flight: 1' and 'version: every' to ensure all PRs flow through the pipeline

### PR DESCRIPTION
**Pull request by tom-meyer**

### CODE REVIEW

File: concourse/pipelines/pr_pipeline.yml [hunk 1]
```diff
@@ -68,12 +68,35 @@ resources:
...
+
   - name: compile_gpdb_centos6
+    max_in_flight: 1
     plan:
     - aggregate:
       - get: gpdb_pr
         trigger: true
-      - get: gpaddon_src 
+        version: every
+        passed: [compile_gpdb_custom_config_centos6]
```
Comment (ashwinstar) [inline concourse/pipelines/pr_pipeline.yml]: Why this kind of dependency ?

File: concourse/pipelines/pr_pipeline.yml [hunk 2]
```diff
@@ -68,12 +68,35 @@ resources:
...
+      input_mapping:
+        gpdb_src: gpdb_pr
+      on_failure:
+        put: gpdb_pr
+        params:
+          path: gpdb_pr
+          status: failure
+
   - name: compile_gpdb_centos6
+    max_in_flight: 1
```
Comment (ashwinstar) [inline concourse/pipelines/pr_pipeline.yml]: What's the logic for `max_in_flight: 1` will solve this problem. That seems too serialized stuff. Not against it but doesn't provide any rational to believe it should solve the problem and why 1 and not like `n`.

Comment (jingyimei) [inline concourse/pipelines/pr_pipeline.yml]: The problem happens to the downstream jobs. If we don't control `max_in_flight : 1` and there are multiple compile jobs in flight, then the downstream jobs will only pickup the latest `gpdb_pr` and `bin_gpdb`.Therefore the downstream ICW and MU tasks might not get the same version of `gpdb_pr` and the corresponding `bin_gpdb`.

File: concourse/pipelines/pr_pipeline.yml (hunk 1 above)
Comment (jingyimei) [inline concourse/pipelines/pr_pipeline.yml]: We will remove the passed constraint. In our previous iteration of solving this we ran into issues where we couldn't guarantee that we don't drop PRs in the middle of the pipeline. We tried to be extra safe this time around. Thanks for getting this to our notice.

File: concourse/pipelines/pr_pipeline.yml (hunk 2 above)
Comment (ashwinstar) [inline concourse/pipelines/pr_pipeline.yml]: Isn't that native logic broken in concourse, if cannot find correct dependency in downstream job ? Why only PR pipeline suffers from same and not main pipeline ?

Comment (cjcjameson) [inline concourse/pipelines/pr_pipeline.yml]: For the basic git resource, or other standard resources, it's a feature that if multiple versions become available all at once, it only runs against the latest. For the PR resource, the expectation that resources represent order-able artifacts is violated.

**Pull request merged**